module/os/linux/zfs/zpl_file.c

```diff
@@ -736,6 +736,15 @@ zpl_putpage(struct page *pp, struct writeback_control *wbc, void *data)
 	return (0);
 }
 
+#ifdef HAVE_WRITEPAGE_T_FOLIO
+static int
+zpl_putfolio(struct folio *pp, struct writeback_control *wbc, void *data)
+{
+	(void) zpl_putpage(&pp->page, wbc, data);
+	return (0);
+}
+#endif
+
```

Comment:
Would it be possible to make this into `zpl_write_cache_pages()` or something like that? Then you could move all of the `write_caghe_pages()` calls into this function along with the `#ifdef` checks. It would be nice to have them in one place to reduce having to do two `#ifdef` checks in `zpl_writepages()`.

Reply:
> Would it be possible to make this into `zpl_write_cache_pages()` or something like that? Then you could move all of the `write_caghe_pages()` calls into this function along with the `#ifdef` checks. It would be nice to have them in one place to reduce having to do two `#ifdef` checks in `zpl_writepages()`.

Done. Please take a look.

Reply:
Looks good to me and thank you.